### PR TITLE
Developed getUserInfoFromJWT function

### PR DESCRIPTION
**Pull request by yashodgayashan**
### FYI
This function will decode the JWT token and return the userId and userType or an error.

### CODE REVIEW

File: core/src/env_platform/utility.bal [hunk 1]
```diff
@@ -300,3 +304,37 @@ function updateAssignments(json[] assignments, json updatedAssignment, string mi
         id = id + 1;
     }
 }
+
+# The `getUserInfoFromJWT` function will decode the JWT token and return the userType and the userId.
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:308]:
```suggestion
# The `getUserInfoFromJWT` function will decode the JWT token and return the userType and the userId of the user.
```

File: core/src/env_platform/utility.bal [hunk 2]
```diff
@@ -300,3 +304,37 @@ function updateAssignments(json[] assignments, json updatedAssignment, string mi
...
+# 
+# + jwt - JWT token.
+# + return - This function will either return the array of strings which includes userId and userType or
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:311]:
```suggestion
# + return - This function will either return an array of strings which includes the userId and userType or 
```

File: core/src/env_platform/utility.bal [hunk 3]
```diff
@@ -300,3 +304,37 @@ function updateAssignments(json[] assignments, json updatedAssignment, string mi
...
+            certificateAlias: "ballerina",
+            trustStore: trustStore
+        }
+    };
+
+    jwt:JwtPayload result = check jwt:validateJwt(check trap <string>split[1], validatorConfig);
+
+    map<json>? customInfo = result?.customClaims;
+    if (customInfo is map<json>) {
+        string UserName = check trap <string>customInfo.name;
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:334]:
```suggestion
        string userName = check trap <string>customInfo.name;
```

File: core/src/env_platform/utility.bal [hunk 4]
```diff
@@ -300,3 +304,37 @@ function updateAssignments(json[] assignments, json updatedAssignment, string mi
...
+        string UserType = check trap <string>customInfo.scope;
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:335]:
```suggestion
        string userType = check trap <string>customInfo.scope;
```

File: core/src/env_platform/utility.bal [hunk 5]
```diff
@@ -300,3 +304,37 @@ function updateAssignments(json[] assignments, json updatedAssignment, string mi
...
+        return [UserName, UserType];
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:336]:
```suggestion
        return [userName, userType];
```

File: core/src/env_platform/utility.bal [hunk 6]
```diff
@@ -300,3 +304,37 @@ function updateAssignments(json[] assignments, json updatedAssignment, string mi
...
+    } else {
+        return error("Authorization failier", message = "Custom claims are not found");
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:338]:
```suggestion
        return error("Authorization failure", message = "Custom claims could not be found.");
```

File: core/src/env_platform/utility.bal [hunk 7]
```diff
@@ -1,3 +1,7 @@
+import ballerina/config as conf;
+import ballerina/crypto;
+import ballerina/jwt;
+import ballerina/stringutils;
 import ballerina/time;
 # The `extractAreaAsJSONArray` function will extract a JSON array
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:6]: Adding a new line after the import.
```suggestion

# The `extractAreaAsJSONArray` function will extract a JSON array 
```

**Comment by yashodgayashan** [comment]
@deshankoswatte Resolved requested changes.

File: core/src/env_platform/utility.bal [hunk 8]
```diff
@@ -300,3 +305,37 @@ function updateAssignments(json[] assignments, json updatedAssignment, string mi
         id = id + 1;
     }
 }
+
+# The `getUserInfoFromJWT` function will decode the JWT token and return the userType and the userId of the user.
+# 
+# + jwt - JWT token.
```
Comment (sanjulamadurapperuma) [inline core/src/env_platform/utility.bal:311]:
```suggestion
# + jwt - The JWT token.
```

File: core/src/env_platform/utility.bal [hunk 9]
```diff
@@ -300,3 +305,37 @@ function updateAssignments(json[] assignments, json updatedAssignment, string mi
...
+# + return - This function will either return an array of strings which includes the userId and userType or 
+# and error.
```
Comment (sanjulamadurapperuma) [inline core/src/env_platform/utility.bal:313]:
```suggestion
# an error.
```

**Comment by yashodgayashan** [comment]
@sanjulamadurapperuma resolved requested changes.

**Comment by yashodgayashan** [comment]
@sanjulamadurapperuma waiting for your review and approval.

**Pull request merged**